plugins/__init__.py

```diff
@@ -1082,29 +1082,12 @@ def validate_type(
 
 
     """
-    if not (
-        fq_type_name.startswith("pydantic.")
-        or fq_type_name.startswith("datetime.")
-        or fq_type_name.startswith("ipaddress.")
-        or fq_type_name.startswith("uuid.")
-    ):
-        return False
-    module_name, type_name = fq_type_name.split(".", 1)
-    module = importlib.import_module(module_name)
-    t = getattr(module, type_name)
-    # Construct pydantic model
-    if validation_parameters is not None:
-        model = pydantic.create_model(
-            fq_type_name, value=(t(**validation_parameters), ...)
-        )
-    else:
-        model = pydantic.create_model(fq_type_name, value=(t, ...))
-    # Do validation
     try:
-        model(value=value)
-    except pydantic.ValidationError:
+        inmanta.validation_type.validate_type(
+            fq_type_name, value, validation_parameters
+        )
+    except pydantic.ValidationError | ValueError:
```

Comment:
I don't think this is supported on Python 3.9. I've never seen it before in this context but I'm assuming it's the union operator that was introduced in 3.10/3.11 (I forget).
```suggestion
    except (pydantic.ValidationError, ValueError):
```